### ISSUE TITLE
tf.linalg.pinv isn't 

### ISSUE THREAD

**Issue by chengdazhi** [tensorflow/tensorflow#61675]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.6.22

### Custom code

No

### OS platform and distribution

Ubuntu 20.04

### Current behavior?

Pinv should give similar gradient to inv. Here's the code snippet and result.
![image](https://github.com/tensorflow/tensorflow/assets/10631563/de9cdb82-1c93-4a18-a79e-95c0b0834ae6)

![image](https://github.com/tensorflow/tensorflow/assets/10631563/cfdadbb9-5506-4181-a868-69830c5ee96b)

### Standalone code to reproduce the issue

```shell
See figures in behavior section.
```

**Comment by SuryanarayanaY** [comment]
Hi @chengdazhi ,

I tested the code with Tf2.12v and I can see the gradients returned by them are almost equal (with precision related differences of order `e-06` range). Also curious why the sample code snippet uses two gradient tapes as there are no second order gradients here. 

Anyways I have tested the code with both ways and results are same except some precision related differences of order 
**e-06** range. Please refer attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/2d8a7aa556651ef701814dce1e69a2de/61675_tf-linalg-inv_pinv.ipynb).

Could you please test with latest versions and let us know the results.

Thanks!

**Comment by chengdazhi** [comment]
Hi @SuryanarayanaY  Thanks for your speedy response. I handcrafted another matrix in [gist](https://colab.research.google.com/gist/chengdazhi/be84e8163d7c5adf810ddd3045d42faa/61675_tf-linalg-inv_pinv.ipynb), and it seems the gradient difference is still pretty large (order e-2).

![image](https://github.com/tensorflow/tensorflow/assets/10631563/e2a9c21f-b2e4-4cbf-99b3-a0b75f979437)

**Comment by SuryanarayanaY** [comment]
@chengdazhi ,

Tested the updated code and found differences upto e-02 order. Attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/5a924806383ff8f32b0fbf40139bb0fc/61675_tf-linalg-inv_pinv-r1.ipynb) for reference. Will escalate to Dev team for their review. Thanks!

**Comment by cantonios** [comment]
I think that's still within expected numerical precision.  The entries where you see the large "error" are near (or exactly) zero in your inverse, which makes them particularly sensitive to small numerical perturbations.  The pseudo inverse needs to do a lot more computation, since it does an SVD, which means small floating-point errors accumulate more.

**Comment by chengdazhi** [comment]
@cantonios  I don't think so. The se3_mat I created is actually pretty well-conditioned. Btw same code seems legit in pytorch.
See the [gist](https://colab.research.google.com/gist/chengdazhi/be84e8163d7c5adf810ddd3045d42faa/61675_tf-linalg-inv_pinv.ipynb#scrollTo=Y-MT7i-T1DxX). I feel there's a bug somewhere.
![image](https://github.com/tensorflow/tensorflow/assets/10631563/0c84d89d-ac34-4c17-b954-34157fe3055c)

**Comment by Venkat6871** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
